[RS-2421] Add doc for configuring additional GatewayClasses (#2004)
tigera/docs: calico/networking/gateway-api.mdx

- `spec.gatewayDeployment.spec.template.spec.containers` allows control over the memory and CPU that Gateway implementation pods can use
- `spec.gatewayControllerDeployment.spec.template.spec.nodeSelector` allows control over where the gateway controller is scheduled.

For full details please see [the `GatewayAPI` reference documentation](../reference/installation/api.mdx#operator.tigera.io/v1.GatewayAPI).

To make use of these customization fields, use `kubectl edit gatewayapi default` to edit the YAML for the `GatewayAPI` resource, and add or modify the customization fields that you require.

### Configure additional GatewayClasses

You may have a need for a tweak or customization that our `GatewayAPI` resource does not yet support.  For example, you may want to create a Gateway with EnvoyProxy config different from that which the Tigera operator provisions, and where the change that you need is not supported by an existing customization field in the `GatewayAPI` resource.

To do this:
- Extract the YAML for the EnvoyProxy and GatewayClass objects that the Tigera operator provisions.

> - Change the EnvoyProxy YAML according to your needs, and create that modified EnvoyProxy object with a different name (than "envoy-proxy-config").

Reviewer: ```suggestion
- Change the EnvoyProxy YAML according to your needs, and rename the modified EnvoyProxy object (not `envoy-proxy-config.yaml`).
```
For clarity.

Author: I'm afraid this isn't a correct suggestion, because it really is creating a new object, not renaming an existing one.

> - Change the GatewayClass YAML to reference the name of the modified EnvoyProxy object, and create that modified GatewayClass with a different name (than "tigera-gateway-class").

Reviewer: Change as above.

Reviewer: Also change "filename" throughout to ``filename``.

Author: First point - as in my previous response.

Second point - actually these are resource names, not file names.  But OK, I think your suggestion still makes sense.
